File: ORB.Core/Models/Threshold.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class Threshold : Entity
    {
        public Threshold()
        {
            RiskTypes = new HashSet<RiskType>();
            ThresholdAppetites = new HashSet<ThresholdAppetite>();
        }

        public int Priority { get; set; }

        public virtual ICollection<RiskType> RiskTypes { get; set; }
        public virtual ICollection<ThresholdAppetite> ThresholdAppetites { get; set; }
    }
}

File: corporate-reporting/src/types/Threshold.ts
import { IRiskType } from "./RiskType";
import { IThresholdAppetite } from "./ThresholdAppetite";
import { IEntity, Entity } from "./Entity";

export interface IThreshold extends IEntity {
    Priority: number;

    RiskTypes?: IRiskType[];
    ThresholdAppetites?: IThresholdAppetite[];
}

export class Threshold extends Entity implements IThreshold {
    public Priority = null;
}